pkg/asset-manager-utils/contracts/RebalancingRelayer.sol

```diff
@@ -51,12 +54,23 @@ contract RebalancingRelayer is IBasePoolRelayer, AssetHelpers {
         return _calledPool == poolId;
     }
 
+    receive() external payable {
+        // Accept ETH transfers only coming from the Vault. This is only expected to happen when joining a pool,
+        // any remaining ETH value will be transferred back to this contract and forwarded back to the original sender.
+        _require(msg.sender == address(vault), Errors.ETH_TRANSFER);
+    }
```

Comment:
This can actually also happen in an exit if the pool holds weth. Because the vault sends eth back to the caller (not `sender`), we'd need to have the same `if balance(this) > 0` check in `exitPool`.